### PR TITLE
feat(app): add URL parameter form prefilling for new items

### PR DESCRIPTION
**Pull request by av1m**
## Scope

What's changed:

* Added **URL parameter form prefilling functionality** for new items.
* Created `parse-prefill-values.ts` utility to extract field values from query parameters in the format `?val[field]=value`.
* Added comprehensive tests for `parse-prefill-values` covering edge cases (arrays, special characters, null/undefined, numeric strings).
* Enhanced `useItem` composable to accept an additional `prefillValues` parameter.

  * Merges schema defaults with prefill values for new items.
  * Updated `refreshItem()` to initialize `item.value` with default + prefill values instead of `null`.
  * Added watcher to reactively update the item when prefill values change.
* Updated `item.vue` to extract prefill values from the route query and pass them into `useItem`.
* Ensured backward compatibility: no changes for existing items being edited.

## Potential Risks / Drawbacks

* Prefill values might unintentionally overwrite schema defaults if both specify the same field.
* URL length limitations could restrict use cases where many fields are prefilled at once.
* Introduces a watcher that may add slight overhead when dealing with large reactive objects.

## Tested Scenarios

* ✅ Prefill with simple field values (`?val[first_name]=Bob`).
* ✅ Handle empty queries gracefully.
* ✅ Ignore non-prefill query parameters (`bookmark`, `sort`, etc.).
* ✅ Handle array values in query params (takes the first value).
* ✅ Support special field names with underscores/numbers.
* ✅ Handle `undefined` and `null` values.
* ✅ Preserve numeric strings as-is.
* ✅ Manual validation in the UI confirms that new forms are prefilled correctly.

## Review Notes / Questions

* Should prefill values always override schema defaults, or should schema defaults take priority if both exist?
* Currently only supports `?val[field]=value`. Should we extend to support direct params (`?first_name=Bob`) as suggested in the discussion?
* Consider future enhancement for `closeOnSave` query parameter (requested in the original feature discussion).

## Checklist

* [x] Added or updated tests
* [x] Documentation PR created https://github.com/directus/docs/pull/491 

Fixes #10232

### CODE REVIEW

**Review by Nitwel — APPROVED** [review]
Changes look very clean and well tested. Thank you for such a great PR and the detailed description on what this is doing. ❤️ 

I will update the docs on this and then this should be ready to be merged.

File: app/src/composables/use-item/index.ts [hunk 1]
```diff
@@ -86,9 +87,33 @@ export function useItem<T extends Item>(
...
 	});
 
-	const defaultValues = getDefaultValuesFromFields(fieldsWithPermissions);
+	const defaultValues = computed(() => {
+		const schemaDefaults = getDefaultValuesFromFields(fieldsWithPermissions).value;
 
-	watch([collection, primaryKey, ...(isRef(query) ? [query] : [])], refresh);
+		// For new items, merge in prefill values from URL parameters
+		if (isNew.value) {
+			const prefill = unref(prefillValues);
```
Comment (Nitwel) [inline app/src/composables/use-item/index.ts:96]: To be able to merge this, we would need some form of basic value parsing/validating here. Basically making sure that if the field is using a certain interface, that this interface either supports a string or that the value will be casted into whatever the interface needs. If the interface is unknown or not supported, we should ignore the default value coming from the query.

**Review by Nitwel — CHANGES_REQUESTED** [review]
After some more discussions, it would make sense to do some basic validation/parsing on these values. Right now, this would only work on interfaces that support string as an input and it breaks on interfaces like the multiple checkboxes.

Happy to answer more questions on how this should look like. Feel free to reach out to me at any time and thanks for the effort you already have put into this.

**Comment by AlexGaillard** [comment]
Hi @av1m it's been a while since Nitwel's comment above, and there's been no movement on this, so I'm going to close this out. If you're happy to make the change requested above then please do so and open a new PR and we'll be happy to take another look 😄
